src/new_index/fetch.rs

```diff
@@ -82,9 +82,24 @@ fn bitcoind_fetcher(
         spawn_thread("bitcoind_fetcher", move || {
             for entries in new_headers.chunks(100) {
                 let blockhashes: Vec<BlockHash> = entries.iter().map(|e| *e.hash()).collect();
-                let blocks = daemon
-                    .getblocks(&blockhashes)
-                    .expect("failed to get blocks from bitcoind");
+
+                let mut attempts = 5;
+                let blocks = loop {
+                    attempts -= 1;
+
+                    match daemon.getblocks(&blockhashes) {
+                        Ok(blocks) => break blocks,
+                        Err(e) => {
```

Comment:
Perhaps this should explicitly check for "Block not found on disk" errors and terminate the thread for other unexpected errors?

Reply:
On a second thought non recoverable errors will panic anyway after 5 attempts, while recoverable errors will leave a trace in the logs. Are we sure we don't want to retry with any error?

Reply:
Anyway, I opened #118 which retry only on "Block not found on disk", one or the other is fine for me

Reply:
Thanks. I generally prefer to fail fast, if both are fine by you then I would go with #118

(Perhaps also worth mentioning that the daemon already detects connection errors and retries for all RPC commands, so the explicit "Block not found on disk" handling here is done on top of that behavior.)